tools/lsp/server/multiplex.toit

```diff
@@ -0,0 +1,96 @@
+// Copyright (C) 2022 Toitware ApS. All rights reserved.
+
+import reader show BufferedReader Reader CloseableReader
+import writer show Writer
+import binary show LITTLE_ENDIAN
+import host.pipe show OpenPipe
+import monitor show Semaphore
+
+/**
+Connection that dispatches the data of the given $OpenPipe to two pipes.
+The $compiler_to_fs and $compiler_to_parser $SimplePipe should be used as a
+  normal $CloseableReader.
+
+If data is produced faster than it is consumed, then the data is buffered. There
+  is now flow-control.
+
+The given incoming $OpenPipe should be the stdout of the C++ compiler. Data
```

Comment:
So if you ever use printing to debug the compiler, we're going to mess with the protocol?

Reply:
Correct.
That was already the case before, though. The compiler published its diagnostics... on stdout, and used a socket to get filesystem information.
Now it uses stdout for both.